mysql-test/main/unique.test

```diff
@@ -23,3 +23,39 @@ DELETE FROM t2 WHERE n = 1;
 DROP TABLE t2;
 
 --source include/test_db_charset_restore.inc
+
+#
+# Start of 11.8 tests
+#
+
```

Comment:
Do these tests add any value? 

I've tried this:

Get the commit before this patch and compile:
```
git checkout 63cbca3fba053b8fb291f7dd7f20adf877bf5869
make 
```
Then, get this patch so that tests are updated:
```
git checkout f4833ceb4133099a501605c9688e34259a0aaa97
``` 
and do NOT compile.
Now, run
```
./mtr unique 
```
and it still passes ....

Reply:
Another question:  Is `unique.test` the right place to put the tests?  I think the problem is not in unique indexes or their handling but rather in the subquery optimizations.